File: other/sorts/comb.js
function combSort(array){
    for (var x = array.length; x > 0; x--){
        for (var y = 0; y < array.length; y++){
            var foo = array[y]
            var bar = array[y + x]
            if (bar && foo > bar){
                array[y] = bar
                array[y + x] = foo
            }
        }
    }
    return array
}

File: other/sorts/comb.rb
def comb_sort array
    array.count.times do |pass|
        foo = array.count - pass
        array.each_index do |index|
            if array[index + foo]
                a = array[index]
                b = array[index + foo]
                if a > b
                    array[index] = b
                    array[index + foo] = a
                end
            end
        end
    end
    array
end